src/vs/editor/contrib/smartSelect/browser/smartSelect.ts

```diff
@@ -311,10 +312,17 @@ CommandsRegistry.registerCommand('_executeSelectionRangeProvider', async functio
 	assertType(URI.isUri(resource));
 
 	const registry = accessor.get(ILanguageFeaturesService).selectionRangeProvider;
+	const configurationService = accessor.get(IConfigurationService);
 	const reference = await accessor.get(ITextModelService).createModelReference(resource);
 
+	const options: SelectionRangesOptions = configurationService.getValue<SelectionRangesOptions | undefined>('editor.smartSelect', { resource }) ?? {
+		// TODO: how to avoid repeating defaults here?
+		selectLeadingAndTrailingWhitespace: true,
```

Comment:
The configuration service should already fallback to the default values. Repeating them here is more for the testing case which has (likely) no defaults

Reply:
Ah, thanks. In that case I think it makes sense to move them to the test service: 1f9e9cd8250e1fadbae3f0ca6ef62ed5c6851757.